src/query/api/v1/handler/prometheus/remote/read_test.go

```diff
@@ -79,11 +87,29 @@ func TestPromReadParsing(t *testing.T) {
 	require.Equal(t, len(r.Queries), 1)
 }
 
+func TestPromFetchTimeoutParsing(t *testing.T) {
+	logging.InitWithCores(nil)
+	ctrl := gomock.NewController(t)
+	storage, _ := m3.NewStorageAndSession(t, ctrl)
+	promRead := &PromReadHandler{
+		engine:          executor.NewEngine(storage, tally.NewTestScope("test", nil), defaultLookbackDuration),
+		promReadMetrics: promReadTestMetrics,
+		timeoutOpts: &prometheus.TimeoutOpts{
+			FetchTimeout: 2 * time.Minute,
+		},
+	}
+
+	req, _ := http.NewRequest("POST", PromReadURL, test.GeneratePromReadBody(t))
+	dur, err := prometheus.ParseRequestTimeout(req, promRead.timeoutOpts.FetchTimeout)
+	require.NoError(t, err)
+	assert.Equal(t, 2*time.Minute, dur)
```

Comment:
Nice!

Can you add a test with a default, and an invalid timeout value?

Reply:
We have these tests in the native package. Not sure we need to duplicate.